gwcs/coordinate_frames.py

```diff
@@ -382,51 +382,53 @@ class TemporalFrame(CoordinateFrame):
 
     Parameters
     ----------
-    axes_order : tuple or int
-        A dimension in the input data that corresponds to this axis.
-    reference_frame : `object`
-        The object to instantiate to represent the time coordinate. Defaults to
-        `astropy.time.Time`. Use partial functions to customise the
-        `~astropy.time.Time` instance.
-    reference_time : `astropy.time.Time` or `None`
-        Reference time, the time of the 0th coordinate. If none the values of
-        the axis are assumed to be valid times.
-    unit : str or units.Unit instance
-        Spectral unit.
+    reference_frame : `~astropy.time.Time`
+        A Time object which holds the time scale and format.
+        If data is provided, it is the time zero point.
+        For relative times `data` should be an empty list.
+    unit : str or `~astropy.units.Unit`
+        Time unit.
     axes_names : str
-        Spectral axis name.
+        Time axis name.
+    axes_order : tuple or int
+        A dimension in the data that corresponds to this axis.
     name : str
         Name for this frame.
     """
 
-    def __init__(self, axes_order=(0,), reference_time=None,
-                 reference_frame=astropy.time.Time, unit=None,
+    def __init__(self, reference_frame, unit=None, axes_order=(0,),
                  axes_names=None, name=None, axis_physical_types=None):
 
+        axes_names = "{}({}; {}".format(reference_frame.format, reference_frame.scale, reference_frame.location)
+
         super().__init__(naxes=1, axes_type="TIME", axes_order=axes_order,
                          axes_names=axes_names, reference_frame=reference_frame,
-                         unit=unit, name=name,
-                         reference_position=reference_time, axis_physical_types=axis_physical_types)
+                         unit=unit, name=name, axis_physical_types=axis_physical_types)
+        self._attrs = {}
+        for a in self.reference_frame.info._represent_as_dict_extra_attrs:
```

Comment:
Does this dict not also have the values of the attrs as well? :man_shrugging:

Reply:
It's actually a tuple
```
t.info._represent_as_dict_extra_attrs                                                                        
Out[8]: 
('format',
 'scale',
 'precision',
 'in_subfmt',
 'out_subfmt',
 'location',
 '_delta_ut1_utc',
 '_delta_tdb_tt')

```